src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/FindAllReferencesHandler.cs

```diff
@@ -114,21 +114,23 @@ public async Task<VSReferenceItem[]> HandleRequestAsync(ReferenceParams request,
                 }
 
                 var razorDocumentUri = RazorLSPConventions.GetRazorDocumentUri(referenceItem.Location.Uri);
-                var mappingResult = await _documentMappingProvider.MapToDocumentRangeAsync(
+                var mappingResult = await _documentMappingProvider.MapToDocumentRangesAsync(
```

Comment:
Would using `RemapLocationsAsync` before this loop be more performant? From my understanding that would do all the remapping in a single request using the individual `Location.Uri` to account for references across multiple files. Am I missing something?

Reply:
BTW, that last commit Fixes: https://github.com/dotnet/aspnetcore/issues/22513 😉

Reply:
That is the goal. With how `RemapLocationsAsync` works currently, it would swallow undefined ranges, which means it is not guaranteed to return the same length of the input array. So we can't use that without a little more refactoring. That will be part of https://github.com/dotnet/aspnetcore/issues/22513. 
Also note, if you look at how RemapLocationsAsync actually is implemented, it doesn't actually do any batching. This is because every location comes with a URI and our MapToDocumentRange requests need to done per file.

Reply:
The more I think about it, the more I feel like FAR doesn't need to use that method as it uses a special `VSReferenceItem[]` type instead of doing what the spec says `Location[]`. I am willing to either backlog or just close https://github.com/dotnet/aspnetcore/issues/22513.

Reply:
Fair enough, thanks makes sense! I've closed out https://github.com/dotnet/aspnetcore/issues/22513.